### PR TITLE
Druid v0.6 compatibility

### PR DESCRIPTION
**Pull request by dimaslv**
These two commits allowed me to set up latest superset with our company's Druid v0.6 cluster. I have tried not to break anything for any newer versions.

### CODE REVIEW

File: superset/models.py [hunk 1]
```diff
@@ -1661,7 +1661,11 @@ def get_druid_version(self):
         endpoint = (
             "http://{obj.coordinator_host}:{obj.coordinator_port}/status"
         ).format(obj=self)
-        return json.loads(requests.get(endpoint).text)['version']
+        ver = json.loads(requests.get(endpoint).text)['version']
```
Comment (mistercrunch) [inline superset/models.py]: I'd prefer an approach where we can specify the DRUID_VERSION as a configuration element. I would have to be associated with the DruidCluster model. Here we'd use the information defined manually and fallback on this current approach when not specified.

File: superset/models.py [hunk 2]
```diff
@@ -1754,18 +1758,38 @@ def generate_metrics(self):
...
-            mt = corrected_type.lower() + 'Avg'
-            name = 'avg__' + self.column_name
-            metrics.append(DruidMetric(
-                metric_name=name,
-                metric_type='avg',
-                verbose_name='AVG({})'.format(self.column_name),
-                json=json.dumps({
-                    'type': mt, 'name': name, 'fieldName': self.column_name})
-            ))
+            if ver > '0':
```
Comment (mistercrunch) [inline superset/models.py]: I'd rather take an approach like `< '0.7'` here

File: superset/models.py [hunk 3]
```diff
@@ -1754,18 +1758,38 @@ def generate_metrics(self):
...
+                metrics.append(DruidMetric(
+                    metric_name=name,
+                    metric_type='postagg',
+                    verbose_name='AVG({})'.format(self.column_name),
+                    json=json.dumps({
+                        'type': 'arithmetic',
+                        'name': name,
+                        'fn': '/',
+                        'fields': [
+                          { 'type': 'fieldAccess', 'fieldName': 'sum__' + self.column_name },
```
Comment (mistercrunch) [inline superset/models.py]: Lint:
no space after `{`
no space before `}`
80 char max

**Comment by mistercrunch** [comment]
Wouldn't it make more sense for this logic to be handled in pydruid?

File: superset/models.py (hunk 1 above)
Comment (dimaslv) [inline superset/models.py]: My programmer skills are quite weak and my knowledge about superset architecture is even more weak. Can you point me on how do I do that? Or maybe you can help me by coding this.

File: superset/models.py (hunk 2 above)
Comment (dimaslv) [inline superset/models.py]: Fixed in 9e5678d

File: superset/models.py (hunk 3 above)
Comment (dimaslv) [inline superset/models.py]: Fixed in 18ba45b

**Comment by dimaslv** [comment]
I don't know anything about pydruid, but actually I don't see any "*avg" aggregations in latest Druid at all, so it's strange that you use such aggregation in you code. Could you explain please?

**Comment by mistercrunch** [comment]
There's been merge conflicts for a long time, feel free to re-open once conflicts are sorted out.

**Comment by dimaslv** [comment]
As I have not received answers to my questions for 5 months, I assume that you are not interested in those patches. Once anyone will speak to me again I will try to resume my work.

**Comment by mistercrunch** [comment]
Sorry it's hard to keep up with so many PRs, hopefully we can ramp up more committers/maintainers soon.

**Pull request closed without merging**